test/test-server/src/emailpassword.ts

```diff
@@ -9,6 +9,8 @@ const { logDebugMessage } = logger(namespace);
 
 const router = Router()
     .post("/signup", async (req, res, next) => {
+        const fdiVersion: string = req.headers["fdi-version"] as string;
```

Comment:
One improvement we could make here:
- add a generic helper function `serializeAndSendResponse(response, req)`
- extract the fdi version inside `serializeAndSendResponse` and pass it to the serialization functions
- apply this wherever possible in the test server (basically replace `res.json`)

I'm thinking you could do something similar when de-serializing.